python/python/glide/async_commands/server_modules/ft_options/ft_profile_options.py

```diff
@@ -0,0 +1,121 @@
+            limited (Optional[bool]): To provide some brief version of the output, otherwise a full verbose output is provided.
+        """
+        self.query = query
+        self.queryType = queryType
+        self.queryOptions = queryOptions
+        self.limited = limited
+
+    @classmethod
+    def from_query_options(
+        cls,
+        query: TEncodable,
+        queryOptions: Union[FtSeachOptions, FtAggregateOptions],
+        limited: Optional[bool] = False,
+    ):
+        """
+        A class method to create FtProfileOptions with FT.SEARCH/FT.AGGREGATE options.
+
+        Args:
+            query (TEncodable): The query that is being profiled. This is the query argument from the FT.AGGREGATE/FT.SEARCH command.
+            queryOptions (Optional[Union[FtSeachOptions, FtAggregateOptions]]): The arguments/options for the FT.AGGREGATE/FT.SEARCH command being profiled.
+            limited (Optional[bool]): To provide some brief version of the output, otherwise a full verbose output is provided.
+        """
+        return cls(query, None, queryOptions, limited)
+
+    @classmethod
+    def from_query_type(
+        cls, query: TEncodable, queryType: QueryType, limited: Optional[bool] = False
+    ):
+        """
+        A class method to create FtProfileOptions with QueryType.
+
+        Args:
+            query (TEncodable): The query that is being profiled. This is the query argument from the FT.AGGREGATE/FT.SEARCH command.
+            queryType (QueryType): The type of query to be profiled.
+            limited (Optional[bool]): To provide some brief version of the output, otherwise a full verbose output is provided.
+        """
+        return cls(query, queryType, None, limited)
+
+    def to_args(self) -> List[TEncodable]:
+        """
+        Get the remaining arguments for the FT.PROFILE command.
+
+        Returns:
+            List[TEncodable]: A list of remaining arguments for the FT.PROFILE command.
+        """
+        args: List[TEncodable] = []
+        if self.queryType:
+            args.extend([self.queryType.value, FtProfileKeywords.QUERY, self.query])
+        elif self.queryOptions:
+            if type(self.queryOptions).__name__ == FtAggregateOptions.__name__:
+                args.extend(
+                    [
+                        QueryType.AGGREGATE.value,
+                        FtProfileKeywords.QUERY,
+                        self.query,
+                    ]
+                    + cast(FtAggregateOptions, self.queryOptions).to_args()
+                )
+            elif type(self.queryOptions).__name__ == FtSeachOptions.__name__:
+                args.extend(
+                    [
+                        QueryType.SEARCH.value,
+                        FtProfileKeywords.QUERY,
+                        self.query,
+                    ]
+                    + cast(FtSeachOptions, self.queryOptions).toArgs()
```

Comment:
```suggestion
                    + cast(FtSeachOptions, self.queryOptions).to_args()
```
rename the method if needed

Reply:
Will update this for the entire VSS module in the cleanup task.